google/cloud/security/common/gcp_type/instance.py

```diff
@@ -136,3 +145,71 @@ def name(self):
             str: name
         """
         return self._path_component('name')
+
+
+# pylint: disable=too-few-public-methods
+class InstanceNetworkInterface(object):
+    """InstanceNetworkInterface Resource."""
+
+    def __init__(self, **kwargs):
+        """Initialize
+
+        Args:
+            kwargs: json from a single instance on the network_interfaces
+        """
+        self.kind = kwargs.get('kind')
+        self.network = kwargs.get('network')
+        self.subnetwork = kwargs.get('subnetwork')
+        self.network_ip = kwargs.get('networkIP')
+        self.name = kwargs.get('name')
+        self.access_configs = kwargs.get('accessConfigs')
+        self.alias_ip_ranges = kwargs.get('aliasIpRanges')
+
+    def __repr__(self):
+        """Repr
+
+        Returns:
+            string: a string for a InstanceNetworkInterface
+        """
+        return 'kind: %s Network: %s subnetwork: %s network_ip %s name %s' \
+            'access_configs %s alias_ip_ranges %s' % (
+                self.kind, self.network, self.subnetwork, self.network_ip,
+                self.name, self.access_configs, self.alias_ip_ranges)
+
+    def __hash__(self):
+        """hash
+
+        Returns:
+            hash: of InstanceNetworkInterface
+        """
+        return hash(self.__repr__())
```

Comment:
the problem I had with hashing a tuple of InstanceNetworkInterface is that some of the variables are dictionaries and some of them are none so you would need to turn it all into strings or write logic to hash everything in the correct order. I thought calling repr was the cleanest way because of this. Do you think a tuple of the strings would be better?